app/javascript/components/student/Editor.tsx

```diff
@@ -111,29 +217,84 @@ export function Editor({
             return
           }
 
-          dispatch({ type: EditorStatus.SUBMISSION_CANCELLED })
+          dispatch({ type: ActionType.SUBMISSION_CANCELLED })
         })
         .finally(() => {
           controllerRef.current = undefined
         })
     },
     [abort, controllerRef]
   )
+  const submit = useCallback(
+    (e: React.MouseEvent<HTMLButtonElement>) => {
+      if (!submission) {
+        return
+      }
+
+      dispatch({ type: ActionType.CREATING_ITERATION })
+
+      fetchJSON(submission.links.submit, { method: 'POST' }).then(
```

Comment:
Make abortable

Reply:
I was thinking about this and I think that we should send an API request back if we ever want to abort this API request. With that said, would it be okay to just ask the user a confirmation message just before they submit? This is to make do without making the request abortable.

(cc @iHiD)

Reply:
Is this aborting a submission? I don't see how we can abort the submission via another API call as we don't have a serialization key, unless we seed the submission uuid in the JS, which we could do, but then we'd need to handle an extra submission state of duplicate uuids (which is fine, but just something to do)

Reply:
I don't want us to prompt users for any reasons that aren't UX-reasons.

Reply:
This is aborting the creation of an iteration. At this point should we just let them submit and not allow them to abort?

Reply:
You want to _abort_ fetching and parsing the response if the user navigates away (this component unmounts).

Reply:
Cool :) I was confused with what you meant. I'll do that.